### PR TITLE
Allow to toggle the use of raw input per window

### PR DESCRIPTION
**Pull request by nathan818fr**
Following my issue #1400.

Features:
- Enable or disable the use raw input for mouse movements (when cursor is disabled) per window
  It's enabled by default and work only if the machine support it
  ~~`int glfwGetWindowUseRawInput(GLFWwindow* handle)`~~
  ~~`void glfwSetWindowUseRawInput(GLFWwindow* handle, int value)`~~
  it use `glfwGetInputMode` / `glfwSetInputMode` with the new mode `GLFW_RAW_INPUT`
- Check weather the machine support raw inputs
  `int glfwRawInputSupported(void)`

### CODE REVIEW

**Comment by nathan818fr** [comment]
I like the idea of `glfwSetInputMode` with a new `GLFW_RAW_INPUT` mode !

**Comment by nathan818fr** [comment]
So I used `glfwSetInputMode` and added doc (I tried to document properly).

File: include/GLFW/glfw3.h [hunk 1]
```diff
@@ -3875,9 +3877,14 @@ GLFWAPI int glfwGetInputMode(GLFWwindow* window, int mode);
  *  GLFW_MOD_CAPS_LOCK bit set when the event was generated with Caps Lock on,
  *  and the @ref GLFW_MOD_NUM_LOCK bit when Num Lock was on.
  *
+ *  If the mode is `GLFW_RAW_INPUT`, the value must be either `GLFW_TRUE` to
+ *  enable the use of raw input, or `GLFW_FALSE` to disable it.  If enabled and
```
Comment (intrigus) [inline include/GLFW/glfw3.h]: You have a double space between "disable it.__If"

Comment (nathan818fr) [reply include/GLFW/glfw3.h]: Yes, I saw that everywhere else there were two spaces after dots, so I did the same.

**Comment by nathan818fr** [comment]
After several days of use under Windows (win 10, x86_64) and Linux (Debian 9, x86_64), I can confirm that it works well!

**Comment by nathan818fr** [comment]
I test this feature since a week on a Minecraft launcher (about 15 000 users per day), no bug report, everything goes well!

**Comment by elmindreda** [comment]
Looks good. f48146ebaad0c0c7181e86679b0ad8118431599d in the [raw-mouse-motion-control](https://github.com/glfw/glfw/compare/raw-mouse-motion-control) branch are my proposed changes.

TL;DR from the summary:
> This renames 'raw input' to 'raw mouse motion' as there are other kinds of raw input.  The update path is restructured to avoid reinitializing all of disabled cursor mode.  Modification of shared state is moved out into shared code.  Raw mouse motion is disabled by default for compatibility.

**Comment by nathan818fr** [comment]
Great!
